File: FE/src/features/payslip/services/payslipService.ts
import { apiClient } from '../../../services/apiClient'
import type { PaginatedResult } from '../../../types/api.types'
import type { CreatePayslipRequest, PayslipResponse } from '../types/payslip.types'

export const payslipService = {
  getAll: (page = 1, size = 10, sort = 'id,desc') => {
    const params = new URLSearchParams({
      page: String(page),
      size: String(size),
      sort,
    })
    return apiClient<PaginatedResult<PayslipResponse>>(`/payslips?${params}`)
  },

  getById: (id: number) => apiClient<PayslipResponse>(`/payslips/${id}`),

  create: (data: CreatePayslipRequest) =>
    apiClient<PayslipResponse>('/payslips', {
      method: 'POST',
      body: JSON.stringify(data),
    }),

  confirm: (id: number) =>
    apiClient<PayslipResponse>(`/payslips/${id}/confirm`, { method: 'POST' }),

  pay: (id: number) => apiClient<PayslipResponse>(`/payslips/${id}/pay`, { method: 'POST' }),

  cancel: (id: number) =>
    apiClient<PayslipResponse>(`/payslips/${id}/cancel`, { method: 'POST' }),
}

File: BE/src/main/java/Manager_vnd/Manager/feature/payslip/PayslipService.java
package Manager_vnd.Manager.feature.payslip;

import java.time.LocalDate;

import Manager_vnd.Manager.dto.PaginatedResult;
import Manager_vnd.Manager.feature.payslip.dto.CreatePayslipRequest;
import Manager_vnd.Manager.feature.payslip.dto.PayslipResponse;
import Manager_vnd.Manager.feature.payslip.dto.UpdatePayslipRequest;

public interface PayslipService {

    PaginatedResult<PayslipResponse> getAll(
            int page, int size, String sort,
            Long workerId, PayslipStatus status,
            LocalDate periodFrom, LocalDate periodTo);

    PayslipResponse getById(long id);

    PayslipResponse create(CreatePayslipRequest request);

    PayslipResponse update(UpdatePayslipRequest request);

    PayslipResponse confirm(long id);

    PayslipResponse pay(long id);

    PayslipResponse cancel(long id);
}
